packages/cms-cli/commands/watch.js

```diff
@@ -28,97 +28,156 @@ const {
 const COMMAND_NAME = 'watch';
+const DESCRIPTION =
+  'Watch a directory on your computer for changes and upload the changed files to the HubSpot CMS';
 
-function configureWatchCommand(program) {
+const action = async ({ src, dest }, command = {}) => {
+  setLogLevel(command);
+  logDebugInfo(command);
+  const {
+    config: configPath,
+    remove,
+    initialUpload,
+    disableInitial,
+    notify,
+  } = command;
+  loadConfig(configPath);
+  checkAndWarnGitInclusion();
+
+  if (disableInitial) {
+    logger.info(
+      'Passing the "--disable-initial" option is no longer necessary. Running "hs watch" no longer uploads the watched directory by default.'
+    );
+  } else {
+    logger.warn(
+      `The "watch" command no longer uploads the watched directory by default. The directory "${src}" was not uploaded.`
+    );
+    logger.warn(
+      'To upload the directory run "hs upload" or add the "--initial-upload" option when running "hs watch".'
+    );
+  }
+
+  if (
+    !(
+      validateConfig() &&
+      (await validatePortal(command)) &&
+      validateMode(command)
+    )
+  ) {
+    process.exit(1);
+  }
+
+  const portalId = getPortalId(command);
+  const mode = getMode(command);
+
+  const absoluteSrcPath = path.resolve(getCwd(), src);
+  try {
+    const stats = fs.statSync(absoluteSrcPath);
+    if (!stats.isDirectory()) {
+      logger.log(`The "${src}" is not a path to a directory`);
+      return;
+    }
+  } catch (e) {
+    logger.log(`The "${src}" is not a path to a directory`);
+    return;
+  }
+
+  if (!dest) {
+    logger.log('A destination directory needs to be passed');
+    return;
+  }
+
+  trackCommandUsage(COMMAND_NAME, { mode }, portalId);
+  watch(portalId, absoluteSrcPath, dest, {
+    mode,
+    cwd: getCwd(),
+    remove,
+    disableInitial: initialUpload ? false : true,
+    notify,
+  });
+};
+
+// Yargs Configuration
+const command = `${COMMAND_NAME} <src> <dest>`;
+const describe = DESCRIPTION;
+const builder = yargs => {
+  addConfigOptions(yargs, true);
+  addPortalOptions(yargs, true);
+  addLoggerOptions(yargs, true);
+  addModeOptions(yargs, { write: true }, true);
+
+  yargs.positional('src', {
+    describe:
+      'Path to the local directory your files are in, relative to your current working directory',
+    type: 'string',
+    demand: true,
+  });
+  yargs.positional('dest', {
+    describe: 'Path in HubSpot Design Tools, can be a net new path',
+    type: 'string',
+    demand: true,
+  });
+  yargs.option('remove', {
+    alias: 'r',
+    describe:
+      'Will cause watch to delete files in your HubSpot account that are not found locally.',
+    type: 'boolean',
+  });
+  yargs.option('initial-upload', {
+    alias: 'i',
+    describe: 'Upload directory before watching for updates',
+    type: 'boolean',
+  });
+  yargs.option('disable-initial', {
+    alias: 'd',
+    describe: 'retrieve most recent log only',
+    type: 'Disable the initial upload when watching a directory (default)',
```

Comment:
Looks like `type` and `describe` are a little off.

Reply:
oh wow hahahah that's what I get for copy/pasting from the docs.  Thanks